remoulade/state/message_state.py

```diff
@@ -0,0 +1,44 @@
+from ..cancel import Cancel
+from .backend import State
+
+
+class Decorator:
+    """Decorator to save the
+    state in a backend
+    """
+    def save(state):
+        def wrapper(decorated):
+            def save_state(*args, **kwargs):
+                key = args[2].message_id
+                args[0].backend.store_result(key,
+                                             State(state),
+                                             args[0].result_ttl)
+            return save_state
+        return wrapper
+
+
+class MessageState(Cancel):
+
+    """Middleware use to storage and update the state
+    of the messages.
+    """
+    def __init__(self, backend, result_ttl=86400):
+        super().__init__(backend=backend)
+        self.backend = backend
+        self.result_ttl = result_ttl
+
+    @Decorator.save(state="Pending")
```

Comment:
Decorator is nice. but I would prefer a simple function,
```
def save_state(self, message, status):
    self.backend.store_result(message.message_id, State(state), self.result_ttl)
```
It is easier to read.